### PR TITLE
[ci-visibility] Link CI Visibility and RUM

### PR DESCRIPTION
**Pull request by juan-fernandez**
## Motivation

Link RUM and CI Visibility through the cypress instrumentation. 

This requires changes to the cypress instrumentation in dd-trace-js: https://github.com/DataDog/dd-trace-js/pull/1706 

Related event format update: DataDog/rum-events-format#49 and DataDog/rum-events-format#50

## Changes

* Read from `window.Cypress`, which is available under cypress tests, and pass info as context. 

## Testing

- [ ] Local
- [ ] Staging
- [x] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/rum-core/src/domain/assembly.ts [hunk 1]
```diff
@@ -96,9 +99,10 @@ export function startRumAssembly(
           date: timeStampNow(),
           service: configuration.service,
           session: {
-            type: syntheticsContext ? SessionType.SYNTHETICS : SessionType.USER,
+            type: syntheticsContext ? SessionType.SYNTHETICS : ciContext ? SessionType.CITEST : SessionType.USER,
```
Comment (bcaudan) [inline packages/rum-core/src/domain/assembly.ts:102]: You need to submit this new type in [rum-event-format](https://github.com/DataDog/rum-events-format).

File: packages/rum-core/src/domain/ciTestContext.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  Cypress?: {
+    env: (key: string) => string | undefined
+  }
+}
+
+export function getCITestContext() {
+  const isCypress = (window as CITestWindow).Cypress
+  const testExecutionId = (window as CITestWindow).Cypress?.env('traceId')
+
+  if (isCypress && typeof testExecutionId === 'string') {
```
Comment (bcaudan) [inline packages/rum-core/src/domain/ciTestContext.ts:11]:
```suggestion
  const testExecutionId = (window as CITestWindow).Cypress?.env('traceId')

  if (typeof testExecutionId === 'string') {
```
no?

File: packages/rum-core/src/domain/assembly.ts (hunk 1 above)
Comment (juan-fernandez) [reply packages/rum-core/src/domain/assembly.ts:102]: will do, thanks!

Comment (juan-fernandez) [reply packages/rum-core/src/domain/assembly.ts:102]: PR in rum-event-format: https://github.com/DataDog/rum-events-format/pull/49

**Comment by juan-fernandez** [comment]
this should be ready for another look @bcaudan

**Comment by juan-fernandez** [comment]
a couple of unit tests are failing, I'm on it

**Comment by juan-fernandez** [comment]
> a couple of unit tests are failing, I'm on it

they should be fixed

File: packages/rum-core/src/domain/assembly.ts [hunk 3]
```diff
@@ -75,6 +77,7 @@ export function startRumAssembly(
   }
 
   const syntheticsContext = getSyntheticsContext()
+  const ciContext = getCiTestContext()
```
Comment (bcaudan) [inline packages/rum-core/src/domain/assembly.ts:80]:
```suggestion
  const ciTestContext = getCiTestContext()
```
for naming consistency

File: packages/rum-core/src/rumEvent.types.ts [hunk 4]
```diff
@@ -678,7 +678,7 @@ export interface CommonProperties {
     /**
      * Type of the session
      */
-    readonly type: 'user' | 'synthetics'
+    readonly type: 'user' | 'synthetics' | 'ci_test'
```
Comment (bcaudan) [inline packages/rum-core/src/rumEvent.types.ts:681]: Did you run `yarn rum-events-format:sync` (cf file header)?
the `rum-events-format` ref should also be updated

Comment (juan-fernandez) [reply packages/rum-core/src/rumEvent.types.ts:681]: good catch

**Review by bcaudan — APPROVED** [review]
Nice!

**Pull request merged**